Path-Check/gaen-mobile
# GAEN-781 - URL Error Message

### Why:

As a user was attempting to check for new exposures, the app was showing to the user that the call succeeded when in fact the app was silently failing.

### This commit:

This PR was used to implement a way for us to wait for the work being done in the background to succeed/fail before showing any kind of result. To do this, we had to add `RXJava` to the project. This enabled us to wait for the `ListenableFuture` to finish doing its work in the background before we continued with our work on the main thread.

Added `ProvideDiagnosisKeyService` to imitate the work that was being done before using `ListenableFuture`. The purpose of this was to give us additional options to manage threads easier and give us additional `callback` functionality.

### How to test:

1. Configure an invalid `DOWNLOAD_BASE_URL` or `DOWNLOAD_PATH`
2. Build Android App and click "**Check For Exposures**"

### Linked issues:

[GAEN-781](https://pathcheck.atlassian.net/jira/software/projects/GAEN/boards/33?selectedIssue=GAEN-781)
